src/payment_store.rs

```diff
@@ -133,16 +128,36 @@ where
 		self.payments.lock().unwrap().contains_key(payment_hash)
 	}
 
-	pub(crate) fn set_status(
-		&self, payment_hash: &PaymentHash, payment_status: PaymentStatus,
-	) -> Result<(), Error> {
+	pub(crate) fn update(
+		&self, payment_hash: &PaymentHash, update_preimage: Option<Option<PaymentPreimage>>,
+		update_secret: Option<Option<PaymentSecret>>, update_amount_msat: Option<Option<u64>>,
+		update_status: Option<PaymentStatus>,
```

Comment:
May be more readable at the call sites if instead of having all these parameters, have some struct with all the the fields that can be modified defaulting to `None`, and only setting the ones that need to be modified.